clorm/orm/core.py

```diff
@@ -2392,11 +2399,14 @@ def _generate_dynamic_predicate_functions(class_name: str, namespace: Dict):
                   "check_complex": check_complex,
                   "args_raw": args_raw,
                   "sign_check_unify": sign_check_unify,
-                  "args_cltopy": args_cltopy}
+                  "args_cltopy": args_cltopy,
+                  "hash_evaluation_self": hash_evaluation_self,
+                  "hash_evaluation_instance": hash_evaluation_instance}
 
-    template = PREDICATE_TEMPLATE.format(pdefn=pdefn)
+    bool_status = not pdefn.is_tuple or len(pdefn) > 0
```

Comment:
Is this by intention that you have changed the __bool__ expression? Before we have just checked for the length

Reply:
Yes the change was intentional. I was trying to think what was most intuitive from a user perspective. 

Most importantly, I think clorm tuples should behave like a normal tuple as much as possible. So an empty tuple should evaluate to false.  

But for a normal fact, even one with no parameters, I'm not sure it makes sense for it to evaluate to false. I think the following would be unintuitive:

```
class P(Predicate):
    pass

p = P()
if not p: 
   print("Feels strange")
```
Anyway, that was the rationale :)

In any case, It's a bit of a boundary case to define predicates with no parameters and clorm is not well suited to this use case. It's not very efficient or intuitive (every `P` instance is equivalent). Ideally it would be better to have a simpler mechanism for representing these cases. I just haven't come up with something that could also be made to fit nicely with the rest of clorm; unifiction, FactBases and the query stuff.